packages/rpc-provider/src/ws/Provider.ts

```diff
@@ -135,13 +135,13 @@ export default class WsProvider implements WSProviderInterface {
     try {
       const WS = await getWSClass();
 
-      this.#websocket = new WS(this.#endpoints[this.#endpointNext]);
+      this.#websocket = new WS(this.#endpoints[this.#endpointIndex]);
       this.#websocket.onclose = this.#onSocketClose;
       this.#websocket.onerror = this.#onSocketError;
       this.#websocket.onmessage = this.#onSocketMessage;
       this.#websocket.onopen = this.#onSocketOpen;
 
-      this.#endpointNext = (this.#endpointNext + 1) % this.#endpoints.length;
+      this.#endpointIndex = (this.#endpointIndex + 1) % this.#endpoints.length;
```

Comment:
Just put this before 138, otherwise that is -1

Reply:
Doh! On it =)